pkg/operator/ceph/controller/cluster_info.go

```diff
@@ -141,6 +141,9 @@ func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.
 		} else {
 			return nil, maxMonID, monMapping, errors.New("failed to find either the cluster admin key or the username")
 		}
+		if len(secrets.OwnerReferences) > 0 {
+			clusterInfo.OwnerInfo = k8sutil.NewOwnerInfoWithOwnerRef(&secrets.GetOwnerReferences()[0], namespace)
```

Comment:
Seems safe to assume there is always an ownerref on this secret, like you said we have had it there for a long time.

Reply:
Yeah. I think it should be safe because of this code above:

```golang
	secrets, err := clusterdContext.Clientset.CoreV1().Secrets(namespace).Get(context, AppName, metav1.GetOptions{})
	if err != nil {
		if !kerrors.IsNotFound(err) {
			return nil, maxMonID, monMapping, errors.Wrap(err, "failed to get mon secrets")
		}
		if ownerInfo == nil {
			return nil, maxMonID, monMapping, ClusterInfoNoClusterNoSecret
		}
```

When a CephCluster is first created and the Secret is initially created and populated (by `initClusterInfo()`/`PopulateExternalClusterInfo()`), the secret shouldn't get created unless there is a CephCluster given to act as the owner.